app/controllers/tags_controller.rb

```diff
@@ -34,14 +34,15 @@ def edit
 
   # Creates a new instance of the tag.
   def create
-    tag_params = params.require(:tag).permit(:name, :description)
+    tag_params = params.require(:tag).permit(:name, :description, :assignment_id)
     new_tag = Tag.new(tag_params.merge(user: @current_user))
 
     if new_tag.save
       if params[:grouping_id]
         grouping = Grouping.find(params[:grouping_id])
         grouping.tags << new_tag
       end
+      new_tag.assessment = Assessment.find(params[:assignment_id])
```

Comment:
Remember that setting an association like this only changes the object in memory, it doesn't save anything to the database. Instead of doing this line, you should change line 38 so that when `new_tag.save` is called, the assessment association is set.